runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/work/refresh/FixedStreamHeartbeatSender.java

```diff
@@ -49,15 +50,24 @@ public static FixedStreamHeartbeatSender create(GetDataStream getDataStream) {
 
   @Override
   public void sendHeartbeats(Heartbeats heartbeats) {
-    if (getDataStream.isShutdown()) {
+    String threadName = Thread.currentThread().getName();
+    try {
+      String backendWorkerToken = getDataStream.backendWorkerToken();
+      if (!backendWorkerToken.isEmpty()) {
+        // Decorate the thread name w/ the backendWorkerToken for debugging. Resets the thread's
+        // name after sending the heartbeats succeeds or fails.
+        Thread.currentThread().setName(threadName + "-" + backendWorkerToken);
```

Comment:
how about we avoid the getName/setName if no backend worker token.
Think you could have threadName start empty, then intialize it here if token was nonempty, and then below set it back only if threadname is nonempty.

Reply:
done used `null` instead of empty string